### PR TITLE
setup project with docker

### PR DESCRIPTION
**Pull request by Manorlds-Eaglespark**
#### What does this PR do?
Setup the project structure, this structure also includes docker, routing and testing environment.

#### Description of Task to be completed?
- Set up a Django (djangorestframework) project
- Make sure the routing works
- Testing environment
- Setup with Docker

#### How should this be manually tested?
1. Install Docker if you do not have it installed on your machine.
2. Use this [link](https://github.com/inshuwa-tech/aggregator-api.git) to clone the project. 
      - Change directory to the cloned project
                  ```cd aggregator-api```
      - Switch to this branch using 
                  ```git checkout ch-project-setup-#4```
3. Run the following commands:
                  ```docker-compose run web python insurance_app/manage.py migrate```
                  ```docker-compose run web python insurance_app/manage.py runserver```
4. Run tests:
                  ```docker-compose run web python insurance_app/manage.py test```
#### What are the relevant Github project issues?
[#4](https://github.com/inshuwa-tech/aggregator-api/issues/4)
#### Screenshots (if appropriate)
Screenshot (23)

### CODE REVIEW

File: insurance_app/life_insurance/views.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from django.core import exceptions
+
+from rest_framework import status
+from rest_framework.generics import RetrieveUpdateAPIView
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from django.shortcuts import render, get_object_or_404
+
+
+class ListCreateLifeInsuranceAPIView(APIView):
```
Comment (mwinel) [inline insurance_app/life_insurance/views.py]: Add a ` docstring ` to define what this class is trying to achieve.

File: insurance_app/life_insurance/views.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+    def get(self, request):
```
Comment (mwinel) [inline insurance_app/life_insurance/views.py]: Add a ` docstring ` to define what this method is trying to achieve. You can also specify what it takes in and what it returns.

Comment (Manorlds-Eaglespark) [reply insurance_app/life_insurance/views.py]: Done. thanks

File: insurance_app/life_insurance/views.py (hunk 1 above)
Comment (Manorlds-Eaglespark) [reply insurance_app/life_insurance/views.py]: Done. Thanks

File: insurance_app/insurance_app/settings.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    },
+]
+
+WSGI_APPLICATION = 'insurance_app.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/2.2/ref/settings/#databases
+
+DATABASES = {
```
[the remark this answers is not in the document]
Comment (Manorlds-Eaglespark) [reply insurance_app/insurance_app/settings.py]: Done, thanks.

File: insurance_app/life_insurance/views.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        """Test method for project setup, to return a simple welcome response"""
```
Comment (mwinel) [inline insurance_app/life_insurance/views.py]: The `test` keyword makes this docstring not appropriate for this method. You should consider redefining it.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FROM python:3
+
+ENV PYTHONUNBUFFERED 1
+RUN mkdir /code
+WORKDIR /code
+COPY . /code/
```
Comment (ManuelDominic) [inline Dockerfile]: @Manorlds-Eaglespark , I suggest you use a reasonable name director like `aggregator_api` for the project to build up the application in order to also reduce these lines of code with just a provision of this code
- `WORKDIR /usr/aggregator_api` which i hope will also be appropriate for our heroku application.
Thank you

File: insurance_app/life_insurance/views.py (hunk 4 above)
Comment (Manorlds-Eaglespark) [reply insurance_app/life_insurance/views.py]: This is not an actual method to be used. It's purpose is only a demostration of facility.

File: Dockerfile (hunk 5 above)
Comment (Manorlds-Eaglespark) [reply Dockerfile]: This is for docker to create an an image, nothing else.

File: insurance_app/insurance_app/settings.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = os.getenv('SECRET_KEY')
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
+
```
Comment (ManuelDominic) [inline insurance_app/insurance_app/settings.py]: @Manorlds-Eaglespark, I also suggest this to be picked from the environmental variable thus for `development` and `production` environments.
Thanks

File: Dockerfile (hunk 5 above)
Comment (ManuelDominic) [inline Dockerfile]: @Manorlds-Eaglespark, I would love to know the value for this line of code?
Thanks

Comment (Manorlds-Eaglespark) [reply Dockerfile]: Docker containerizes the application by running it in it's own environment. So when it begins, it copies the whole project to the new directory we just created, installs the dependencies and runs it from there isolated environment. I recommend you read more about docker to fully grasp the concept. Thanks.

**Pull request merged**